src/lib/core/hooks/visualizations.ts

```diff
@@ -39,3 +50,115 @@ export function useVizConfig<ConfigType>(
 
   return [vizConfig, updateVizConfig];
 }
+
+/**
+ * To be used in conjunction with InputVariables.providedOptionalVariable
+ *
+ * Sets up a "listener" to monitor changes in providedVariableDescriptor
+ * and if the existing vizConfig[inputName] variable is not null
+ * (when the user has "None" selected in the radio buttons),
+ * and if the new variable is compatible with the constraints, then
+ * update the vizConfig[inputName] to the provided variable.
+ *
+ * If vizConfig[inputName] is not null and the variable does not meet the
+ * constraints, then set vizConfig[inputName] to undefined (which has
+ * the effect of moving the radio button to "None".
+ *
+ * Also enqueues a snackbar. Make sure you are in a child of a SnackbarProvider.
+ *
+ * Doesn't return anything!
+ */
+export function useProvidedOptionalVariable<ConfigType>(
+  optionGetter:
```

Comment:
Should this just be an `isEnabled: boolean` prop?

Reply:
I think if it was a named prop, it would be OK to use a boolean, but these are unnamed, so booleans are a bit opaque. Leaving it as-is.